### PR TITLE
Enable Logging

### PR DESCRIPTION
**Pull request by Kuldip-Nanda**

### CODE REVIEW

File: lib/ibm/cloud/sdk/logging.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module IBM
+  module Cloud
+    module SDK
+      class << self
+          attr_writer :logger
```
Comment (Fryguy) [inline lib/ibm/cloud/sdk/logging.rb:5]: Minor, 2 space indent.

File: lib/ibm/cloud/sdk/version.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module IBM
   module Cloud
     module SDK
-      VERSION = "0.1.6"
+      VERSION = "0.1.7"
```
Comment (jaywcarman) [inline lib/ibm/cloud/sdk/version.rb:4]: FYI: I think the usual practice is to push version updates directly instead of through a PR.

**Review by jaywcarman — APPROVED** [review]
Travis even decided to run for you!

File: lib/ibm/cloud/sdk/version.rb (hunk 2 above)
Comment (agrare) [reply lib/ibm/cloud/sdk/version.rb:4]: Correct,  this lets us group whatever PRs we decide should go into a version together

File: lib/ibm/cloud/sdk/power_iaas.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@ def initialize(region, guid, token, crn, tenant)
           @region = region
           @token  = token
           @tenant = tenant
+          RestClient.log = log
```
Comment (agrare) [inline lib/ibm/cloud/sdk/power_iaas.rb:18]: Where is this `log` variable coming from?  I don't see it in the initialize options.  Did you mean `RestClient.log = logger` maybe?  That would be available if you did `include Logging` in this class.

Comment (Kuldip-Nanda) [reply lib/ibm/cloud/sdk/power_iaas.rb:18]: My bad, was done in a hurry, It should be
RestClient.log = logger
and I should include Loggi9ng.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 All notable changes to the gem ibm-cloud-sdk-ruby will be documented here.
 
+## v0.1.7 - 2020-09-28
```
Comment (Fryguy) [inline CHANGELOG.md:4]: Can you change this to `Unreleased - 2020-09-28` (we will change it when we cut the release, as the version may not be 0.1.7 if we need a minor or major instead)

File: lib/ibm/cloud/sdk/power_iaas.rb [hunk 5]
```diff
@@ -15,6 +16,7 @@ def initialize(region, guid, token, crn, tenant)
           @region = region
           @token  = token
           @tenant = tenant
+          RestClient.log = Logging.logger
```
Comment (Fryguy) [inline lib/ibm/cloud/sdk/power_iaas.rb:19]: I don't think Logging.logger exists unless you define it as `def self.logger` up in the Logging module.

Comment (Fryguy) [reply lib/ibm/cloud/sdk/power_iaas.rb:19]: I'm actually surprised tests are passing

Comment (agrare) [reply lib/ibm/cloud/sdk/power_iaas.rb:19]: I think you want to `include Logging` then you can just use `logger`

File: lib/ibm/cloud/sdk/power_iaas.rb [hunk 6]
```diff
@@ -1,3 +1,4 @@
+require_relative 'logging'
 module IBM
```
Comment (agrare) [inline lib/ibm/cloud/sdk/power_iaas.rb:2]: Should add a newline here

File: lib/ibm/cloud/sdk/power_iaas.rb [hunk 7]
```diff
@@ -15,7 +16,7 @@ def initialize(region, guid, token, crn, tenant)
           @region = region
           @token  = token
           @tenant = tenant
-          RestClient.log = IBM::Cloud::SDK::Logging.logger
+          RestClient.log = Logging.logger
```
Comment (agrare) [inline lib/ibm/cloud/sdk/power_iaas.rb:19]: With this applied
```
       class PowerIaas < BaseService
+        include Logging
+
         # Create an API Client object for the Power IaaS service
         #
         # @param region [String] the IBM Power Cloud instance region
@@ -16,7 +19,7 @@ module IBM
           @region = region
           @token  = token
           @tenant = tenant
-          RestClient.log = Logging.logger
+          RestClient.log = logger
```

I'm getting
```
>> IBM::Cloud::SDK::PowerIaas.new("a", "b", "c", "d", "e")
Traceback (most recent call last):
        3: from /home/grare/adam/src/manageiq/ibm-cloud-sdk-ruby/lib/ibm/cloud/sdk/power_iaas.rb:22:in `initialize'
        2: from /home/grare/adam/src/manageiq/ibm-cloud-sdk-ruby/lib/ibm/cloud/sdk/logging.rb:14:in `logger'
        1: from /home/grare/adam/src/manageiq/ibm-cloud-sdk-ruby/lib/ibm/cloud/sdk/logging.rb:9:in `logger'
NameError (uninitialized constant IBM::Cloud::SDK::NullLogger)
```

Comment (Fryguy) [reply lib/ibm/cloud/sdk/power_iaas.rb:19]: Seems like we need specs for PowerIaas class.

**Review by agrare — APPROVED** [review]

**Pull request merged**